nulib/dc-api-v2
# Update `template.yaml` and `make build` to support build-time-only resources

This PR adds some logic to the build/deploy process that allows the SAM template to differ between dev-time and deploy-time. This is done by adding a new kind of comment (`#*`) to lines that should be part of the build but not included during dev, and having the build command strip those comments out before building.

It also updates dependencies and test invocation to get rid of some deprecation warnings. Most notably, we no longer need AWS SDK v2, which saves nearly 9,000 lines among the the three `package-lock.json` files alone.

To test:

1. `make clean`
2. Run API locally with `make serve`
    - Make sure you can use the API without a bunch of layer rebuilds in between, even after modifying a function
3. Build API with `make build`
    - Make sure `template.yaml` is _not_ modified afterward (i.e., put back the way it was)
    - Check `.aws-sam/build/apiDependencies` to make sure all the runtime dependencies are in there
    - Check other node functions under `.aws-sam/build/` to make sure they do _not_ have dependencies
      - Note: This applies only to the main API functions in `node`, not the functions in `lambdas`
      - Also Note: There may be directories under these functions' `node_modules` directory, but they should be empty

Remember to `make clean` again (or at least `rm -rf .aws-sam`) to make sure you don't leave the build behind.
